akka-http/src/main/scala/akka/http/scaladsl/server/directives/ParameterDirectives.scala

```diff
@@ -125,7 +125,10 @@ object ParameterDirectives extends ParameterDirectives {
       extractRequestContext flatMap { ctx ⇒
         import ctx.executionContext
         import ctx.materializer
-        handleParamResult(paramName, fsou(ctx.request.uri.query().get(paramName)))
+        Try(ctx.request.uri.query()) match {
+          case Success(query) ⇒ handleParamResult(paramName, fsou(query.get(paramName)))
+          case Failure(t)     ⇒ reject(MalformedRequestContentRejection("The request's query string is invalid", t))
```

Comment:
I was thinking about the same thing, so looked at usages of the rejection and we don't really add the malformed thing in the message... Though now I realise that in other cases that would have been the entity payload, so likely too big to include in log. 

In this case though, so it's shorter at least should be... sounds good to include it :)